fe/fe-core/src/main/java/org/apache/doris/rewrite/CompoundPredicateWriteRule.java

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.rewrite;
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.BoolLiteral;
+import org.apache.doris.analysis.CompoundPredicate;
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.common.AnalysisException;
+
+import java.util.List;
+
+/**
+ * add Rewrite CompoundPredicates 'OR' 'AND' 'NOT' Rule
+ * It can be applied to pre-analysis expr trees and therefore does not reanalyze
+ * the transformation output itself.
+ * Examples:
+ * OR:
```

Comment:
The comments here are duplicated with the following ones. I'd like to suggest removing this, just use the below comments.
```
        /*
         *  'OR' 'AND' rewrite rule
         *  case true AND expr ==> expr
         *  case expr AND true ==> expr
         *  case false Or expr ==> expr
         *  case expr Or false ==> expr
         *
         *  case false AND expr ==> false
         *  case expr AND false ==> false
         *  case true Or expr ==> true
         *  case expr Or true ==> true
         *
         */
```

Reply:
ok